nimbus.fml.yaml

```diff
@@ -68,8 +68,220 @@ features:
+          OPEN_SYNC_SETTINGS: firefox://settings/sync
+          OPEN_POCKET_SETTINGS: https:///getpocket.com/settings?fxa={fxa-token}
+    defaults:
+      - channel: developer
+        value: {
+          "messages": {
+            "my-viewpoint-survey": {
+              "text": "Love Firefox? Fill in our survey!",
+              "action": "https://surveyprovider.com/survey-id/{client-id}",
+              "trigger": [],
+              "max-display-count": 5,
+              "style": "warning"
+            }
+          },
+
+          "message-under-experiment": "my-viewpoint-survey"
+        }
+      - channel: developer
+        value: {
+          "messages": {
+            "private-tabs-auto-close": {
+              "action": "OPEN_SETTINGS_PRIVATE_TABS",
+              "text": "Sharing your phone? Autoclosing private tabs is for you!",
+              "style": "warning",
+              "trigger": [
+                  "new-user",
+                  "first-private-tabs-opened"
+              ],
+              "max-display-count": 5
+            }
+          },
+
+          "message-under-experiment": "private-tabs-auto-close"
+        }
+      - channel: developer
+        value: {
+          "triggers": {
+            "ireland": "'IE' in locale"
+          },
+
+          "styles": {
+            "irish-green": {
+              "background-color": "green",
+              "text-color": "dark-green",
+              "button-background": "foo",
+              "button-text-color": "very-green",
+              "priority": 50
+            }
+          },
+
+          "actions": {
+            "OPEN_GDPR_SETTINGS": "firefox://settings/gdpr"
+          },
+
+          "messages": {
+            "eu-tracking-protection-for-ireland": {
+              "action": "OPEN_GDPR_SETTINGS",
+              "text": "GDPR has you covered. Firefox has GDPR covered",
+              "style": "irish-green",
+              "button-label": "Click me",
+              "trigger": [
+                  "new-user",
+                  "ireland"
+              ],
+              "max-display-count": 5
+            }
+          },
+
+          "message-under-experiment": "eu-tracking-protection-for-"
+        }
+
+
+
 types:
-  objects: {}
+  objects:
+    MessageData:
+      description: >
+        An object to describe a message. It uses human
+        readable strings to describe the triggers, action and
+        style of the message as well as the text of the message
+        and call to action.
+      fields:
+        action:
+          type: String
+          description: >
+            A URL of a page or a deeplink.
+            This may have substitution variables in.
+          # This should never be defaulted.
+          default: empty_string
+        text:
+          type: Text
+          description: "The message text displayed to the user"
+          # This should never be defaulted.
+          default: empty_string
+        button-label:
```

Comment:
@jhugman  Could you give me a hand :), I'm having some trouble trying to get the `button-label` into the `MessageData`class in Kotlin.
```
> Task :app:nimbusFeaturesDebug FAILED
Nimbus FML generating Kotlin
manifest  nimbus.fml.yaml
class      org.mozilla.fenix.nimbus.FxNimbus
channel    developer
R.class    org.mozilla.fenix.R
Error: Validation Error: Mismatch between type BundleText("") and default null

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:nimbusFeaturesDebug'.
> Process 'command '/app/build/bin/nimbus/nimbus-fml'' finished with non-zero exit value 1

```

Reply:
This looks like one of the bugs in FML that is fixed by https://github.com/mozilla/application-services/pull/4867 .

The current work around is to use `String` instead of `Text`.

Reply:
Thanks for the suggestion, I gave it a try, but unfortunately I still getting a similar issue,  I updated the `type` to `String` and the `default` to `"default"`:

<img width="734" alt="image" src="https://user-images.githubusercontent.com/773158/157451574-ed80346c-1ec0-47df-b458-edbf4683f80b.png">


```
> Task :app:nimbusExperimenter
Nimbus FML generating JSON
manifest   /fenix/nimbus.fml.yaml
outputFile /fenix/.experimenter.json

> Task :app:nimbusFeaturesDebug FAILED
Nimbus FML generating Kotlin
manifest   /fenix/nimbus.fml.yaml
class      org.mozilla.fenix.nimbus.FxNimbus
channel    developer
R.class    org.mozilla.fenix.R
Error: Validation Error: Mismatch between type String and default null

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:nimbusFeaturesDebug'.
> Process 'command '/fenix/app/build/bin/nimbus/nimbus-fml'' finished with non-zero exit value 1


```